### PR TITLE
Fix previous Prettier path in `draft-blog-post` script

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

https://github.com/prettier/prettier/pull/13436#issuecomment-1490921414

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
I think it caused by exports in package.json. This self-referencing behavior is annoying.

I find I can't use `npx prettier` to run stable version. Do you have this problem? Don't know if we can make it work.

**Comment by fisker** [comment]
It seems we can't? https://nodejs.org/api/packages.html#self-referencing-a-package-using-its-name

**Comment by sosukesuzuki** [comment]
Yes, it is annoying...

```
> npx prettier --version
3.0.0-alpha.6
```

I don't know how to avoid this problem without using directly paths to `node_modules`

File: scripts/draft-blog-post.js [hunk 1]
```diff
@@ -22,7 +22,8 @@ const introFile = path.join(changelogUnreleasedDirPath, "blog-post-intro.md");
 if (!fs.existsSync(introFile)) {
   fs.copyFileSync(introTemplateFile, introFile);
 }
-const previousVersion = require("prettier/package.json").version;
+const previousVersion =
+  require("../node_modules/prettier/package.json").version;
```
Comment (fisker) [inline scripts/draft-blog-post.js:26]: Let's go with this for now, but throw if they equal.

File: scripts/draft-blog-post.js [hunk 2]
```diff
@@ -22,7 +22,8 @@ const introFile = path.join(changelogUnreleasedDirPath, "blog-post-intro.md");
...
 const version = require("../package.json").version.replace(/-.+/, "");
```
Comment (fisker) [inline scripts/draft-blog-post.js:27]: Can you also use `semver` to parse it and use major/minor/patch to get the version?

Comment (sosukesuzuki) [reply scripts/draft-blog-post.js]: Yes, but why? I've pushed new commit [2f5c1dd](https://github.com/prettier/prettier/pull/14636/commits/2f5c1dd00993d8c1cde9adec446b1768f0fc2026). Please check it if it follows your imagine.

Comment (fisker) [reply scripts/draft-blog-post.js]: Just to be safer.

**Comment by fisker** [comment]
I mean use `semver` instead of replace with RegExp. https://github.com/prettier/prettier/pull/14636/commits/dec34f6be364ce562caad277ded8a5ca48ac0f2a

**Comment by fisker** [comment]
Moved version check to `replaceVersions` so patch release can get checked too.

**Comment by fisker** [comment]
You can merge if it's fine to you.

**Pull request merged**